torchmetrics/metric.py

```diff
@@ -203,37 +209,59 @@ def forward(self, *args: Any, **kwargs: Any) -> Any:
         Input arguments are the exact same as corresponding ``update`` method. The returned output is the exact same as
         the output of ``compute``.
         """
-        # add current step
+        # check if states are already synced
         if self._is_synced:
             raise TorchMetricsUserError(
-                "The Metric shouldn't be synced when performing ``update``. "
+                "The Metric shouldn't be synced when performing ``forward``. "
                 "HINT: Did you forget to call ``unsync`` ?."
             )
 
-        # global accumulation
-        self.update(*args, **kwargs)
+        if self.full_state_update or self.dist_sync_on_step:
```

Comment:
Just a question: Should we encapsulate the different paths to separate functions so that they can be modified independent of each other?

Reply:
done :)